tests/test_modeling_roberta.py

```diff
@@ -198,6 +198,69 @@ def create_and_check_for_causal_lm(
         result = model(input_ids, attention_mask=input_mask, token_type_ids=token_type_ids, labels=token_labels)
         self.parent.assertEqual(result.logits.shape, (self.batch_size, self.seq_length, self.vocab_size))
 
+    def create_and_check_decoder_model_past_large_inputs(
+        self,
+        config,
+        input_ids,
+        token_type_ids,
+        input_mask,
+        sequence_labels,
+        token_labels,
+        choice_labels,
+        encoder_hidden_states,
+        encoder_attention_mask,
+    ):
+        config.is_decoder = True
+        config.add_cross_attention = True
+        model = RobertaForCausalLM(config=config).to(torch_device).eval()
+
+        # make sure that ids don't start with pad token
+        mask = input_ids.ne(config.pad_token_id).long()
+        input_ids = input_ids * mask
```

Comment:
for `roberta` Position numbers begin at `padding_idx+1` as stated in the doc string here 
https://github.com/huggingface/transformers/blob/f4432b7e01dc46008fb823096d884bdc2861b49c/src/transformers/models/roberta/modeling_roberta.py#L1360-L1364



so the input shouldn't start with pad token. If `past_key_values` are used and if the input starts with pad token then the position ids are off by one. i.e

If it starts with pad and `past_key_values` is not used, position_ids become  `[1, 2, 3, 4...]`

but if `past_key_values` are used then for the second position the position id will be 3 since we add past_key_values_length to it and there is no way to know if the first token was pas token.